onnxruntime/core/graph/graph_utils.cc

```diff
@@ -54,18 +54,20 @@ static bool OutputEdgeProvidesImplicitInput(const Graph& graph, const GraphEdge&
 
 /** Checks if new_output_name can be used to replace removed_output_name in the subgraph input.
     If there is an existing NodeArg in a subgraph that implicitly consumes removed_output_name, it is not safe. */
-static bool CanUpdateImplicitInputNameInSubgraph(Node& node,
+static bool CanUpdateImplicitInputNameInSubgraph(const Node& node,
                                                  const std::string& removed_output_name,
                                                  const std::string& new_output_name) {
-  for (auto& attr_subgraph_pair : node.GetAttributeNameToMutableSubgraphMap()) {
-    Graph& subgraph = *attr_subgraph_pair.second;
+  if (!node.ContainsSubgraph())
+    return true;
+
+  for (auto& subgraph : node.GetSubgraphs()) {
```

Comment:
shouldnt this be a `const auto* subgraph`?

Reply:
it's a gsl::not_null<> instance, so it's a reference. I could put that instead of 'auto' to make it clearer though.